### ISSUE TITLE
Remove dependency on `create-react-context`

### ISSUE THREAD

**Issue by isc30** [CJY0208/react-activation#230]
Hey, really useful library!

I noticed it uses `create-react-context` as a dependency, and `create-react-context` only supports up to `react: ^16.0.0`.
Do you think it's possible to remove that dependency and use the normal 'createContext` from React?

**Comment by CJY0208** [comment]
is there any problem right now?

`create-react-context` is used between versions `16.0.0` and `16.3.0` to support `React.createContext` which is only supported on `16.3.0`, it works as follows

- between versions `16.0.0` and `16.3.0`: use implementation methods of itself
- `^16.3.0`: use `React.createContext` directilly

![image](https://user-images.githubusercontent.com/18415774/194979137-3c436313-5e09-4fbd-b19b-befd2dcd5a23.png)

**Comment by isc30** [comment]
yeah, as it's marked a direct dependency instead of optional dependency,
PNPM warns when using React18 as it's not supported by create-react-context.

I think making it an optional dependency could solve it
